AutoPk with find and query macros

When `id` is an `AutoPk<i32>`, using `find!` and `query!` I see

```
error[E0308]: mismatched types
  --> examples/user-table/tests/unmigrate.rs:56:16
   |
56 |     let post = query!(Post, id == 1).load(&connection).await.unwrap();
   |                ^^^^^^^^^^^^^^^^^^^^^
   |                |
   |                expected `&AutoPk<i32>`, found `&{integer}`
   |                arguments to this method are incorrect
   |
   = note: expected reference `&AutoPk<i32>`
              found reference `&{integer}`
note: method defined here
  --> /Users/jayvdb/rust/butane/butane_core/src/query/fieldexpr.rs:54:16
   |
54 |     binary_op!(eq, std::cmp::PartialEq<U>, Eq);
   |                ^^
   = note: this error originates in the macro `butane::filter` which comes from the expansion of the macro `query` (in Nightly builds, run with -Z macro-backtrace for more info)

```

--- a/butane/tests/query.rs
+++ b/butane/tests/query.rs
@@ -1,6 +1,6 @@
 use butane::db::{Connection, ConnectionAsync};
 use butane::query::BoolExpr;
-use butane::{colname, filter, find, find_async, query, Many};
+use butane::{colname, filter, find, find_async, model, query, AutoPk, Many, SqlVal};
 use butane_test_helper::*;
 use butane_test_macros::butane_test;
 #[cfg(feature = "datetime")]
@@ -276,4 +276,33 @@ async fn offset(conn: ConnectionAsync) {
     assert_eq!(posts.len(), 2);
     assert_eq!(posts[0].title, "Sir Charles");
     assert_eq!(posts[1].title, "The Tiger");
+}
+
+#[model]
+#[derive(PartialEq, Debug)]
+struct HasAutopk {
+    id: AutoPk<i64>,
+    text: String,
+}
+impl HasAutopk {
+    fn new(text: &str) -> Self {
+        HasAutopk {
+            id: AutoPk::default(),
+            text: text.to_string(),
+        }
+    }
+}
+
+#[butane_test]
+async fn query_autopk_by_integer(conn: ConnectionAsync) {
+    let mut val1: HasAutopk = HasAutopk::new("first");
+    let mut val2 = HasAutopk::new("second");
+    val1.save(&conn).await.unwrap();
+    val2.save(&conn).await.unwrap();
+    assert_eq!(
+        HasAutopk::fields().id().eq(&1i64),
+        BoolExpr::Eq("id", butane::query::Expr::Val(SqlVal::BigInt(1)))
+    );
+    let query_results = query!(HasAutopk, id == 1).load(&conn).await.unwrap();
+    assert_eq!(&val1, query_results.first().unwrap());
 }

--- a/butane_core/src/autopk.rs
+++ b/butane_core/src/autopk.rs
@@ -88,6 +88,16 @@ impl<T: PrimaryKeyType> PartialEq for AutoPk<T> {
     }
 }
 
+impl<T: PrimaryKeyType> PartialEq<T> for AutoPk<T> {
+    fn eq(&self, other: &T) -> bool {
+        if !self.is_valid() || !other.is_valid() {
+            false
+        } else {
+            self.inner.as_ref().eq(&Some(other))
+        }
+    }
+}
+
 impl<T: PrimaryKeyType> FieldType for AutoPk<T> {
     const SQLTYPE: SqlType = T::SQLTYPE;
     /// Reference type. Used for ergonomics with String (which has

--- a/butane_core/src/query/mod.rs
+++ b/butane_core/src/query/mod.rs
@@ -21,7 +21,7 @@ pub use fieldexpr::{DataOrd, FieldExpr, ManyFieldExpr};
 type TblName = Cow<'static, str>;
 
 /// Abstract representation of a database expression.
-#[derive(Clone, Debug)]
+#[derive(Clone, Debug, PartialEq)]
 pub enum Expr {
     /// A column, referenced by name.
     Column(&'static str),
@@ -34,7 +34,7 @@ pub enum Expr {
 }
 
 /// Abstract representation of a boolean expression.
-#[derive(Clone, Debug)]
+#[derive(Clone, Debug, PartialEq)]
 pub enum BoolExpr {
     True,
     Eq(&'static str, Expr),
@@ -84,7 +84,7 @@ pub struct Order {
     pub column: &'static str,
 }
 
-#[derive(Clone, Debug)]
+#[derive(Clone, Debug, PartialEq)]
 pub enum Join {
     /// Inner join `join_table` where `col1` is equal to
     /// `col2`
@@ -95,7 +95,7 @@ pub enum Join {
     },
 }
 
-#[derive(Clone, Debug)]
+#[derive(Clone, Debug, PartialEq)]
 pub struct Column {
     table: Option<TblName>,
     name: &'static str,

